crates/miden-testing/src/kernel_tests/tx/test_note.rs

```diff
@@ -679,3 +685,49 @@ pub fn test_timelock() {
     let tx_context = TransactionContextBuilder::new(account).tx_inputs(tx_inputs).build();
     tx_context.execute().unwrap();
 }
+
+#[test]
+fn test_public_key_as_note_input() {
+    let mock_public_key = PublicKey::new([ZERO, ONE, Felt::new(2), Felt::new(3)]);
+    let rpo_component = RpoFalcon512::new(mock_public_key);
+
+    let mock_seed = Digest::from([ZERO, ONE, Felt::new(2), Felt::new(3)]).as_bytes();
+    let target_account = AccountBuilder::new(mock_seed)
+        .with_component(BasicWallet)
+        .with_component(rpo_component)
+        .build_existing()
+        .unwrap();
+
+    let mock_seed =
+        Digest::from([Felt::new(4), Felt::new(5), Felt::new(6), Felt::new(7)]).as_bytes();
+    let sender_account = AccountBuilder::new(mock_seed)
+        .with_component(BasicWallet)
+        .build_existing()
+        .unwrap();
+
+    let serial_num =
+        RpoRandomCoin::new([ONE, Felt::new(2), Felt::new(3), Felt::new(4)]).draw_word();
+    let tag = NoteTag::from_account_id(target_account.id(), NoteExecutionMode::Local).unwrap();
+    let metadata = NoteMetadata::new(
+        sender_account.id(),
+        NoteType::Public,
+        tag,
+        NoteExecutionHint::always(),
+        Default::default(),
+    )
+    .unwrap();
+    let vault = NoteAssets::new(vec![]).unwrap();
+    let note_script =
+        NoteScript::compile("begin nop end", TransactionKernel::testing_assembler()).unwrap();
+    let recipient = NoteRecipient::new(
+        serial_num,
+        note_script,
+        NoteInputs::new(vec![ZERO, ONE, Felt::new(2), Felt::new(3)]).unwrap(),
```

Comment:
The point of the test is that this word is the same as mock_public_key, right? Could we make a single variable out of this and then add a comment to explain that this is used both as pub key as well as note input? Just to make it easier to understand.